silkworm/db/datastore/domain_range_latest_query.hpp

```diff
@@ -58,6 +58,11 @@ struct DomainRangeLatestQuery {
     static_assert(std::same_as<Key1, Key2>);
     using Key = Key1;
 
+    using Word1 = typename TKeyDecoder2::Word;
+    using Word2 = typename TValueDecoder2::Word;
+    static_assert(std::same_as<Word1, Word2>);
+    using Word = Word1;
+
```

Comment:
```suggestion
    using Word = snapshots::Decoder::Word;
```
Since snapshots::Decoder::Word = BytesOrByteView in the snapshots::Decoder interface, and TKeyDecoder2 / TValueDecoder2 implement a snapshots::DecoderConcept - they need to inherit Decoder interface, so TKeyDecoder2::Word is always BytesOrByteView. I suggest to keep it simple and use snapshots::Decoder::Word directly.